### PR TITLE
Combine `set` actions

### PR DESCRIPTION
**Pull request by CoSNaYe**
1. Add "Combine `set` actions" rule
2. Add `Under consideration` and `References` sections
3. add rule: Prefer single quote or backtick for String.  …
4. add rule: Add `propTypes` to every components
5. add rule: Use multiple lines when we need to pass more than one props.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +64,12 @@ A coding guide line for React + Redux project @WORD UP
   * Use [normalizr](https://github.com/paularmstrong/normalizr) if needed.
   * [Normalizing State Shape](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape)
 
+#### Combine `set` actions
+  * Instead of using `state.set('key1', value1).set('key2', 'value2')`, use `withMutations` to wrap multiple actions into one update. Since Immutable.js will perform re-arranging in all the processes and retain all intermediate states.
```
Comment (foodaka) [inline README.md:89]: I think we should give an actual example here of what it should look like

File: README.md [hunk 2]
```diff
@@ -51,6 +64,12 @@ A coding guide line for React + Redux project @WORD UP
...
+
+#### Refrain from using `.toJS()`
```
Comment (foodaka) [inline README.md]: good

File: README.md (hunk 1 above)
Comment (CoSNaYe) [reply README.md:89]: good idea, added in 1ffe8dc

**Comment by CoSNaYe** [comment]
Slack integration test.

**Pull request merged**